### PR TITLE
Alessio add discount field

### LINKED ISSUE #255: Add field to the form for discount code
## Feature request
Add field for discount code, on insert( or on click button 'apply discount' ) the code will be verified and validated and the discount will be calculated.

The coupon can also be inserted during an upgrade

### PR DESCRIPTION
**Pull request by cpalessio96**
### All Submissions:

*   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
*   [x] Does your code has proper inline documentation? 

### Changes proposed in this Pull Request:
Add fields for discount code

Closes #255 .

### CODE REVIEW

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage decreased per file
===========================
- includes/Dashboard/Settings.php  -1
         

Complexity increasing per file
==============================
- src/stripe/Components/PriceTable.js  1
- includes/Api/ApiPricingPlan.php  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

File: assets/css/stripeStyle.css [hunk 1]
```diff
@@ -232,7 +232,7 @@ table tr {
 	margin: 0 -0.5rem;
 }
 
-.button-bar button {
+.button-bar button, .coupon-button {
```
Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:235]: Codacy found an issue: [Expected newline after "," (selector-list-comma-newline-after)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708)

Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:235]: Codacy found an issue: [Expected selector ".button-bar button" to come before selector "button:hover:enabled" (no-descending-specificity)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- includes/Api/ApiPricingPlan.php  1
- src/stripe/Components/PriceTable.js  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

File: assets/css/stripeStyle.css (hunk 1 above)
Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:235]: Codacy found an issue: [Expected selector ".button-bar button" to come before selector "button:hover:enabled" (no-descending-specificity)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708)

Comment (GiuseppeIuculano) [inline assets/css/stripeStyle.css:235]: Codacy found an issue: [Expected newline after "," (selector-list-comma-newline-after)](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/stripe/Components/PriceTable.js  1
- includes/Api/ApiPricingPlan.php  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Api/ApiPricingPlan.php  1
- src/stripe/Components/PriceTable.js  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

File: src/stripe/SubscriptionForm.js [hunk 2]
```diff
@@ -172,6 +174,14 @@ function App() {
 			});
 	};
 
+	const updatePrice = (currentPrice) => {
+		if (currentPrice) {
+			setDiscountedPrice(currentPrice);
+		} else {
+			setDiscountedPrice(false);
+		}
+	};
```
Comment (HyprCoder) [inline src/stripe/SubscriptionForm.js]:
```suggestion
       const newPrice = currentPrice || false;
		setDiscountedPrice(currentPrice);
	};
```

**Review by HyprCoder — CHANGES_REQUESTED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Api/ApiPricingPlan.php  1
- src/stripe/Components/PriceTable.js  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

**Review by HyprCoder — APPROVED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/stripe/Components/PriceTable.js  1
- includes/Api/ApiPricingPlan.php  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Api/ApiPricingPlan.php  1
- src/stripe/Components/PriceTable.js  1
         

Clones added
============
- includes/Api/ApiPricingPlan.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5954708&bid=19516407)

**Pull request merged**